src/main/java/com/uci/utils/cache/controller/CaffeineCacheController.java

```diff
@@ -23,48 +25,76 @@
-	}
+        try {
+            if (authorizationKey.equals(authorizationHeader)) {
+                jsonNode = mapper.readTree("{\"id\":\"api.content.cache\",\"ver\":\"3.0\",\"ts\":\"2021-06-26T22:47:05Z+05:30\",\"responseCode\":\"OK\",\"result\":{}}");
+                JsonNode resultNode = mapper.createObjectNode();
+                cache.asMap().keySet().forEach(key -> {
+                    String cacheName = key.toString();
+                    ((ObjectNode) resultNode).put(cacheName, cache.getIfPresent(cacheName).toString());
+                });
+                ((ObjectNode) jsonNode).put("result", resultNode);
+                return ResponseEntity.ok(resultNode);
+            } else {
+                Map<String, Object> map = new HashMap<>();
+                map.put("message", "Unauthorized. Invalid secure key.");
+                map.put("status", "failed");
+                return new ResponseEntity<>(map, HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception ex) {
+            log.error("CaffeineCacheController:getAll: Error while getting cache : " + ex.getMessage());
+            Map<String, Object> map = new HashMap<>();
+            map.put("message", ex.getMessage());
+            map.put("status", "failed");
+            return new ResponseEntity<>(map, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
 
-	/**
-	 * call this to invalidate all cache instances
-	 */
-	@DeleteMapping(path = "/removeAll")
-	public void removeAll() {
-		cache.asMap().keySet().forEach(key -> {
-			removeCache(key.toString());
-		});
-	}
+    /**
+     * call this to invalidate all cache instances
+     */
+    @DeleteMapping(path = "/removeAll")
+    public ResponseEntity<Object> removeAll(@RequestHeader(name = "Authorization") String authorizationHeader) {
+        try {
+            if (authorizationKey.equals(authorizationHeader)) {
+                cache.asMap().keySet().forEach(key -> {
+                    removeCache(key.toString());
+                });
+                log.info("All cache removed success");
+                Map<String, Object> map = new HashMap<>();
+                map.put("message", "Cache removed success");
+                map.put("status", "success");
+                return new ResponseEntity<>(map, HttpStatus.OK);
+            } else {
+                Map<String, Object> map = new HashMap<>();
+                map.put("message", "Unauthorized. Invalid secure key.");
+                map.put("status", "failed");
+                return new ResponseEntity<>(map, HttpStatus.UNAUTHORIZED);
+            }
+        } catch (Exception ex) {
+            log.error("CaffeineCacheController:removeAll: Error while removing cache : " + ex.getMessage());
+            Map<String, Object> map = new HashMap<>();
+            map.put("message", ex.getMessage());
+            map.put("status", "failed");
+            return new ResponseEntity<>(map, HttpStatus.INTERNAL_SERVER_ERROR);
+        }
+    }
```

Comment:
Hey these changes have been already incorporated by a separate PR. Could you rebase and update the PR?